lib/mongoid/contextual/memory.rb

```diff
@@ -285,13 +285,23 @@ def pick(*fields)
       #   context.tally(:name)
       #
       # @param [ String | Symbol ] field Field to tally.
+      # @param [ Boolean ] :splat_arrays Whether to tally array
+      #   member values individually. Default false.
       #
       # @return [ Hash ] The hash of counts.
-      def tally(field)
-        return documents.each_with_object({}) do |d, acc|
-          v = retrieve_value_at_path(d, field)
-          acc[v] ||= 0
-          acc[v] += 1
+      def tally(field, splat_arrays: false)
```

Comment:
I'm open to calling this parameter `unwind`, which matches the MongoDB name, but I think splat_arrays may be more intuitive for Rubyists.